lib/auth/mocku2f/mocku2f.go

```diff
@@ -0,0 +1,173 @@
+package mocku2f
+
+/* Mock U2F device for testing.
+ * This is not a complete implementation of U2F keys.
+ * In particular, the key always returns a dummy key handle and doesn't differentiate between key handles
+ */
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/tstranex/u2f"
+)
+
+type MockU2fKey struct {
+	privatekey *ecdsa.PrivateKey
+	cert []byte
+	counter uint32
+}
+
+const keyHandle = "asdf"
+
+func decodeBase64(s string) ([]byte, error) {
+	for i := 0; i < len(s)%4; i++ {
+		s += "="
+	}
+	return base64.URLEncoding.DecodeString(s)
+}
+
+func encodeBase64(buf []byte) string {
+	s := base64.URLEncoding.EncodeToString(buf)
+	return strings.TrimRight(s, "=")
+}
+
+func selfSignPublicKey(keyToSign *ecdsa.PublicKey) (cert []byte, err error) {
+	caPrivateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return []byte{}, err
+	}
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			Organization: []string{"Test CA"},
+		},
+		NotBefore: time.Now(),
```

Comment:
I think that won't be very useful. Currently we are skipping the certificate check during registration. The reason we still generate a self-signed certificate here is because the message format requires it. If we do enable certificate verification, the self-signed certificate won't verify anyway. The best we can do here is make the time range longer.